vllm/entrypoints/cli/serve.py

```diff
@@ -109,6 +106,29 @@ def signal_handler(signum, frame):
     signal.signal(signal.SIGTERM, signal_handler)
     signal.signal(signal.SIGINT, signal_handler)
 
+    if parallel_config.distributed_node_rank_within_dp > 0:
```

Comment:
Would be good to add a comment here explaining that this is for running "headless workers" mode.